### DISCUSSION TITLE
大模型在生成回复的时候，如何生成答案的引用来源呢？

### DISCUSSION THREAD

**Discussion by minmie** [THUDM/GLM-4#678] [Q&A] [answered]
![cd248267461fe1553c8baf487061dc6](https://github.com/user-attachments/assets/2c2a8b3d-eaef-4261-a7bd-72bd7479f0d7)

如图，模型回答的时候，会标记回答的来源和依据，这种如何实现呢？

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
这个是RAG + 工程，并在UI上显示。有的大模型，比如Longcite-glm4-9b 这个模型也支持自己能返回对应的索引，不需要RAG。
但是显示效果是UI显示的，不是大模型能力，你需要自己开发UI
